streamy.py

```diff
@@ -75,3 +56,60 @@ def stream(fp, json_lines=False, **kwds):
             obj = json.loads(json_str, **kwds)
             start_pos += end_pos
             yield obj
+
+
+def _stream_without_seek(fp, kwds):
+    buf = ''
+    # XXX: alternatively could read a limited size buffer
+    for l in fp:
+        buf += l
+        while buf:
+            try:
+                obj = json.loads(buf, **kwds)
+            except ValueError as exc:
+                end_pos = _get_end_pos(exc)
+                if end_pos is None:
+                    if exc.msg.startswith('Expecting value') \
+                       and not buf.strip():
+                        buf = ''
+                        continue
+                    else:
```

Comment:
You don't need the `else`.  Actually, I think this whole thing might be better as:

    if end_pos is not None:
        raise
    # Continue here without the else.

Reply:
I don't get how that's equivalent